[caffe2] Add AVX512 support for box_cox operator

efiks opened the pull request:
Summary:
Reuse templetized implementation of box_cox caffe2 operator.
* Duplicate .cc file of AVX2
* change intrinsics functions to use AVX512 instructions
* override templates
* extend the caller to use new methods
* guard AVX512 with a gflag to allow smooth transition

Differential Revision: D67433457

hl475 reviewed (approved):
Please make sure both external and internal CI signals are green before landing

efiks commented:
@pytorchbot label "topic: not user facing"

efiks committed:
[caffe2] Add AVX512 support for box_cox operator (#143627)

Summary:

topic: not user facing

Reuse templetized implementation of box_cox caffe2 operator.
* Duplicate .cc file of AVX2
* change intrinsics functions to use AVX512 instructions
* override templates
* extend the caller to use new methods
* guard AVX512 with a gflag to allow smooth transition
* fixed rowprune unit tests

Reviewed By: hl475

Differential Revision: D67433457
